lib/compact_index/versions_file.rb

```diff
@@ -15,80 +15,34 @@ def contents(gems = nil, args = {})
 
       raise ArgumentError, "Unknown options: #{args.keys.join(", ")}" unless args.empty?
 
-      out = File.read(@path)
-      out << parse_gems(gems) if gems
-      out
+      File.read(@path).tap do |out|
+        out << gem_lines(gems) if gems
+      end
     end
 
     def updated_at
       created_at_header(@path) || Time.at(0).to_datetime
     end
 
-    def update_with(gems)
```

Comment:
Is this approach scalable? Given that `create` [adds a new line for each version](https://github.com/bundler/compact_index/issues/17) and rubygems we get 2000-2500 versions each week.